style_guide/README.md

```diff
@@ -0,0 +1,200 @@
+   create a new pull request.  Click on the "compare across forks" link to
+   enable the pull request to use your fork.  Set the "base repository" to
+   `IMAP-Science-Operations-Center` and "base" to `dev`.  Set the "head
+   repository" to the `imap_processing` repository under your personal fork
+   and "compare" to your feature branch. If the feature branch is still under
+   development, you can click the "Convert to draft" button under the
+   "Reviewers" section, or add a "[WIP]" at the beginning of the pull request
+   title to signify that the pull request is not ready to be merged.
+
+10. Assign at least one reviewer to the pull request.  They will review your
+    pull request and either accept the request or ask for additional changes.
+    If additional changes are needed, iterate through steps (6) and (7) until
+    you and the reviewer(s) are satisfied.
+
+11. Once the pull request has been accepted, you can merge the pull request and
+    delete the feature branch.
+
+
+### Making hotfixes
+
+As shown in the diagram above, sometimes hotfixes need to be directly to the
+`main` branch. The workflow for this scenario is as follows:
+
+1. Assuming steps (1) through (3) in the previous section are already completed,
+   create a new branch named `hotfix-<description>` off of the `main` branch,
+   and commit any necessary changes following the nominal `git add`/`git commit`
+   cycle.
+2. Push the hotfix branch to your personal fork, and open two separate pull
+   requests: one that merges the hotfix branch into the
+   `IMAP-Science-Operations-Center` `imap_processing` `main` branch, and one
+   that merges the hotfix branch into the `IMAP-Science-Operations-Center`
+   `imap_processing` `dev` branch.
+3. For each of these pull requests, proceed with the nominal review & merge
+   process described in steps (9) and (10) in the previous section.
+
+
+### Keeping your fork updated
+
+You can keep your personal fork up-to-date with the
+`IMAP-Science-Operations-Cetner` `imap_processing` repository by fetching and
+pulling the `upstream` remote:
+
+    git checkout dev
+    git fetch upstream dev
+    git pull upstream/dev
+
+### Collaborating on someone else's fork
+
+To contribute to a branch on another person's personal fork, add a new `remote`
+that points to their fork, and use the nominal workflow for contributing:
+
+    git remote add <username> <remote url>
+    git fetch <username>
+    git checkout -b <branchname> <username>/<branchname>
+    # Make some changes via add/commit cycle
+    git push <username> <branchname>
+
+Python Coding
+-------------
+
+`imap_processing` code shall adhere to the [PEP8](https://www.python.org/dev/peps/pep-0008/)
+conventions save for the following exceptions:
+
+ - Lines of code need not be restricted to 79 characters.  However, it is
+encouraged to break up excessively long lines into several lines if it benefits
+the overall readability of the code
```

Comment:
Add something about 88 characters since we are using black? Really end users don't have a choice with black how things are formatted.